feat: control filter sidebar visibility from querystring

jvanveen opened the pull request:
#### Checklist
- [x] Functionality works & meets specifications
- [x] Code reviewed
- [x] Code unit/integration/system tested
-  User documentation updated
- [x] Conventional commits (squash if needed)
-  No warnings during install
-  Updated javascript typing

jvanveen commented:
De knop in de toolbarview verdwijnt met https://github.com/molgenis/molgenis-frontend/pull/535, dus die is ongewijzigd in deze pr.

molgenis-jenkins commented:
:star: PR Preview available on https://preview-frontend-pr-537.dev.molgenis.org

connoratrug commented:
![](https://github.trello.services/images/mini-trello-icon.png) [add filter show/hide status to the route query (jeroen)  https://github.com/molgenis/molgenis-frontend/pull/537](https://trello.com/c/kPWwLJbM/418-add-filter-show-hide-status-to-the-route-query-jeroen-https-githubcom-molgenis-molgenis-frontend-pull-537)

connoratrug reviewed (commented):
When i close the filter bar , when open the bar , when click close again nothing happens.
When i click close a second time the bar does close

connoratrug commented on packages/data-explorer/src/router.ts:
```diff
@@ -1,34 +1,26 @@
-        if (to.params.entity) {
-          entity = to.params.entity
-        }
+const routes = [
```
should the file name be changed to routes ?

jvanveen replied:
That makes sense indeed. Changed

connoratrug commented on packages/data-explorer/src/types/RouteQuery.ts:
```diff
@@ -3,5 +3,6 @@ export type RouteQuery = {
     size?: string, // page size integer as a string
     sort?: string, // name of sort column, prefix with '-' for reverse sort
     filter?: string // hashed filter object
+    filterbar?: '0' | '1' // filter sidebar
```
why is an int (as string) instead of a bool ( as string) , why filterbar and not 'showfilters' or 'isfilersshown' or something like that

jvanveen replied:
I suppose its a matter a of personal preference. I find an int easier to parse to a boolean and to toggle states between. Same for the naming of the variable; I would rather not use verbs, because the other parameters are nouns.

connoratrug commented on packages/data-explorer/src/views/FiltersView.vue:
```diff
@@ -90,6 +89,13 @@ export default Vue.extend({
       })
+    },
+    toggleFilterbar () {
+      let filterbarState = this.$route.query.filterbar ? parseInt(this.$route.query.filterbar, 10) : 1
```
why the parsing instead of a simple bool

jvanveen replied:
The querystring cannot be typed differently than string (afaik). Flipping an int seems more straightforward to me than switching between 'true' and 'false' strings.

connoratrug reviewed (dismissed):
the need to click two times needs to be fixed i thing , the rest is working nicely

molgenis-jenkins commented:
:star: PR Preview available on https://preview-frontend-pr-537.dev.molgenis.org

jvanveen commented:
Renamed the filterbar variable to hideSidebar (same as the state variable), using 'false'/'true' now instead of '0'/'1'. Renamed router.ts to 'routes.ts'. Fixed tests.

molgenis-jenkins commented:
:star: PR Preview available on https://preview-frontend-pr-537.dev.molgenis.org

molgenis-jenkins commented:
:star: PR Preview available on https://preview-frontend-pr-537.dev.molgenis.org

molgenis-jenkins commented:
:star: PR Preview available on https://preview-frontend-pr-537.dev.molgenis.org
